### ISSUE TITLE
Receiver not registered exception.

### ISSUE THREAD

**Issue by ubarua123** [razorpay/razorpay-android-sample-app#11]
``` java
Caused by java.lang.IllegalArgumentException: Receiver not registered:        com.razorpay.SmsAgent$if@42818048
       at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:667)
       at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1582)
       at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:489)
       at com.razorpay.SmsAgent.ˎ(SourceFile:82)
       at com.razorpay.OTPElf.paymentFlowEnd(SourceFile:272)
       at com.razorpay.CheckoutActivity.onDestroy(SourceFile:182)
       at android.app.Activity.performDestroy(Activity.java:5420)
       at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1140)
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:3551)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:3582)
       at android.app.ActivityThread.access$1400(ActivityThread.java:139)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1263)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:136)
       at android.app.ActivityThread.main(ActivityThread.java:5086)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

Got this on a live app.

**Comment by shashankmehta** [comment]
@ubarua123 This is the second report we've got for the same error. After the first report we tried debugging it but so far we've not been able to reproduce the issue. The error comes up if the SMS Broadcast receiver is not registered but the SDK tries to deregister it. However, there doesn't seem to be a a code flow where a receiver could not be registered and SDK would still try to deregister it.

We have still worked on a pre-emptive fix for this and we have shared the SDK with the folks who reported it first. We will also be adding this to our next SDK version that is going to be rolled out early next week. I would suggest that once that version is released, please update the SDK in your app.

I apologise for the issue. I wish I could give a better response at this stage but since we are not able to reproduce the bug on our end, we are not able to find the exact fix yet.

**Comment by hariramsvnit** [comment]
@ubarua123 We have released v1.0.1 of sdk, I would highly recommend you to upgrade razorpay sdk. We have done a pre-emptive fix for this issue also. You can refer documentation from [here](https://docs.razorpay.com/v1/page/android-sdk-v10x)

**Comment by sunil456** [comment]
W/System.err: java.lang.IllegalArgumentException: Receiver not registered: com.razorpay.AutoReadOtpHelper@77f1142
        at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1429)
        at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1543)
        at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:664)
        at com.razorpay.CheckoutPresenterImpl.cleanUpOnDestroy(CheckoutPresenterImpl.java:705)
        at com.razorpay.b__J_.onDestroy(BaseCheckoutActivity.java:178)
W/System.err:     at android.app.Activity.performDestroy(Activity.java:8048)
        at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1334)
        at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4926)
        at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4970)
        at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:44)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)

**Comment by vivekshindhe** [comment]
@sunil456 this has been fixed in version 1.6.15. Please update and check.
